sdk/src/client/secret/mod.rs

```diff
@@ -567,14 +567,15 @@ where
             .required_address(slot_index, protocol_parameters.committable_age_range())?
             .ok_or(crate::client::Error::ExpirationDeadzone)?;
 
-        let required_address = if let Address::Restricted(restricted) = &required_address {
-            restricted.address()
-        } else {
-            &required_address
+        // Convert restricted and implicit addresses to Ed25519 address, so they're the same entry in `block_indexes`.
+        let required_address = match required_address {
+            Address::Restricted(restricted) => restricted.address().clone(),
+            Address::ImplicitAccountCreation(implicit) => Address::Ed25519(*implicit.ed25519_address()),
```

Comment:
Do you mind switching both pls? So they are in order

Reply:
Done